packages/@o3r/core/schematics/service/index.ts

```diff
@@ -54,26 +54,45 @@ export function ngGenerateService(options: NgGenerateServiceSchematicsSchema): R
       packageName = JSON.parse(tree.read(path.resolve(destination, rec, 'package.json'))!.toString()).name?.split('/')[0] || destination;
     }
 
-    const templateSource = apply(url('./templates'), [
-      template({
-        ...strings,
-        ...options,
-        featureName,
-        currentServiceIndex,
-        currentFixtureJasmineIndex,
-        currentFixtureJestIndex,
-        serviceName: strings.capitalize(strings.camelize(options.name + ' ' + options.featureName)),
-        packageName
-      }),
+    const templateData = {
+      ...strings,
+      ...options,
+      featureName,
+      currentServiceIndex,
+      currentFixtureJasmineIndex,
+      currentFixtureJestIndex,
+      serviceName: strings.capitalize(strings.camelize(options.name + ' ' + options.featureName)),
+      packageName
+    };
+
+    const baseTemplateSource = apply(url('./templates/base'), [
+      template(templateData),
       renameTemplateFiles(),
       move(destination)
     ]);
+
+    const rules = [mergeWith(baseTemplateSource, MergeStrategy.Overwrite)];
+
+    ['jest', 'jasmine-core'].forEach(testFramework => {
```

Comment:
it is the second time we use this way to determine the test system, maybe we can provide an helper for it in `@o3r/schematics`:

```typescript
export type AvailableTestFrameworsks = 'jest' | 'jasmine';

const testPackageMaps: Record<AvailableTestFrameworsks, string[]> = {
  'jest': ['jest'],
  'jasmine': ['jasmine-core']
}

export function getTestFramework(workspaceOptions?: WorkspaceOptions): AvailableTestFrameworsks | undefined {
  if (workspaceOptions?.testFramework) {
    return workspaceOptions.testFramework;
  }
  return Object.entries(testPackageMaps)
    .find(([_, deps]) => {
      deps.some((dep) => {
        try {
          return !!require.resolve(`${dep}/package.json`);
        } catch {
          return false;
        }
      });
    })?.[0];
}

Reply:
Could you please point where we are using it the first time? I couldn't find it anywhere.

Reply:
I guess in `library` and `application` generators (and it will be needed in `component`, `services` and `store` :))

Reply:
I am pretty sure this is not used anywhere else yet.
On top of that:
1. your solution returns only one occurrence. As we discussed, to remain backward compatible with our private library, we should generate fixtures for both frameworks if the 2 are installed in the project.
2. `testFramework` does not exist on `WorkspaceLayout`

Reply:
the helper function will come with https://github.com/AmadeusITGroup/otter/pull/613